### PR TITLE
[IA-1683] CIS-harden GCE custom images

### PR DESCRIPTION
**Pull request by kyuksel**
As explained in the [decision log](https://broadworkbench.atlassian.net/wiki/spaces/IA/pages/239009797/2020-02-11+Choosing+Base+Image+for+Custom+GCE+VM+Images), we had decided to harden our GCE images ourselves according to the CIS guidelines. This PR incorporates the hardening into the script that gets run when creating the images.

An example of a resulting image can be found [here](https://console.cloud.google.com/compute/imagesDetail/projects/broad-dsde-dev/global/images/leo-custom-gce-image-200224-11-17?project=broad-dsde-dev).

### CODE REVIEW

**Comment by kyuksel** [comment]
Bond failure in fiab-start  🤷‍♂ 

jenkins retest

File: jenkins/gce-custom-images/prepare_custom_leonardo_gce_image.sh [hunk 1]
```diff
@@ -124,6 +135,57 @@ add-apt-repository \
   ${os_dist_code_name:?} \
   ${os_dist_release_channel:?}"
 
+# Do some set-up in preparation for image hardening
+
+# Install dpkg-dev so we can use dpkg-architecture down below
+apt-get install -y -q dpkg-dev
+
+# Install Python (needed to install Ansible)
```
Comment (rtitle) [inline jenkins/gce-custom-images/prepare_custom_leonardo_gce_image.sh]: Do we need to install python 3.7 from source or can we just use the python version that comes with stretch?

**Comment by adrazhi** [comment]
@rtitle @kyuksel Just a reminder that we'd want to maintain our own playbooks which @ssyms  is going to work on when she's back (she's on vacation this week).  This set of somewhere-else-built-modules were provided just for @kyuksel to test in the meantime and make sure that the hardened images wouldn't be broken or interfere with some functionality.

**Comment by rtitle** [comment]
OK, sounds good

**Review by rtitle — APPROVED** [review]

**Review by adrazhi — APPROVED** [review]

**Pull request merged**